extensions/statuspage-monitor/src/service.tsx

```diff
@@ -0,0 +1,149 @@
+import { AllIncidents, ComponentGroup, Incident, Summary } from "./lib/types";
+import { useState, useEffect } from "react";
+import { getAllIncidents, getSummary } from "./lib/api";
+import { Action, ActionPanel, Icon, List, showToast } from "@raycast/api";
+import { capitalizeFirstLetter, formatDate, groupComponents, iconForIndicator } from "./lib/util";
+
+export default function ServiceStatusList({ id }: { id: string }) {
+  const [isLoading, setLoading] = useState(true);
+  const [summary, setSummary] = useState<Summary>();
+  const [allIncidents, setAllIncidents] = useState<AllIncidents>();
+  const [showDetail, setShowDetail] = useState(false);
+
+  useEffect(() => {
+    if (!summary) fetchServiceData();
+  }, []);
+
+  async function fetchServiceData() {
+    setLoading(true);
+
+    const fetchSummary = await getSummary(id).catch(() => null);
+    const fetchAllIncidents = await getAllIncidents(id).catch(() => null);
+
+    if (!fetchSummary || !fetchAllIncidents)
+      return showToast({ title: "Connection Failed", message: "Failed to fetch data." });
+
+    setSummary(fetchSummary);
+    setAllIncidents(fetchAllIncidents);
+    setLoading(false);
+  }
+
+  function selectionChangeHandler(id?: string) {
+    setShowDetail(
+      !!(id && (summary?.incidents?.find((i) => i.id === id) || allIncidents?.incidents?.find((i) => i.id === id)))
+    );
+  }
+
+  const hasOngoingIncidents = !!summary?.incidents?.length;
+  const hasComponents = !!summary?.components?.length;
+  const groupedComponents = summary?.components && groupComponents(summary?.components);
+
+  return (
+    <List isLoading={isLoading} isShowingDetail={showDetail} onSelectionChange={selectionChangeHandler}>
```

Comment:
I think it would be nice in this list to have an action to open incident or component url in the browser